### PR TITLE
Add forceScope opts to useUnit in solidjs

### LINKED ISSUE #766: Isomorphic `useUnit` in `effector-solid`
Same as https://github.com/effector/effector/issues/765

### PR DESCRIPTION
**Pull request by usmanyunusov**
### Conventions
- [x] Please check your messages [against the guidelines](https://cbea.ms/git-commit/) if not, perform [interactive rebase](https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history)
- [x] [Link an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) your pull request closes or relates

Fix #766

### CODE REVIEW

File: src/solid/scope.ts [hunk 1]
```diff
@@ -11,12 +11,12 @@ import {useStoreMapBase, useUnitBase} from './lib/base'
 import {useGate as commonUseGate, createGateImplementation} from './lib/gate'
 import type {Gate} from './index.h'
 
-const ScopeContext = createContext<Scope | null>(null)
+const ScopeContext = createContext(null as Scope | null)
 export const {Provider} = ScopeContext
 
-function getScope() {
+export function getScope(forceScope?: boolean) {
```
Comment (igorkamyshev) [inline src/solid/scope.ts:17]: In my opinion, it is better not to export this function from public module, let's move it somewhere inside the package?

**Review by igorkamyshev — COMMENTED** [review]
Thanks for a contribution 💙

File: src/solid/scope.ts (hunk 1 above)
Comment (usmanyunusov) [reply src/solid/scope.ts:17]: Maybe take move everything the file `scope.ts` into an `ssr.ts` file. Leave only public methods in the `ssr.ts`.

File: src/solid/scope.ts [hunk 2]
```diff
@@ -58,8 +58,9 @@ export function useGate<Props>(
 
   createEffect(() => commonUseGate(ForkedGate(), props))
 }
-export function useUnit(shape) {
-  return useUnitBase(shape, getScope())
+export function useUnit(shape, opts?: {forceScope?: boolean}) {
+  const scope = getScope(opts?.forceScope)
```
Comment (AlexandrHoroshih) [inline src/solid/scope.ts:62]: I think that `/scope` version of bindings should have `forceScope` enabled by default 🤔 

I suggest to change it like that:
```suggestion
  const scope = getScope(opts?.forceScope ?? true)
```

File: src/solid/scope.ts (hunk 1 above)
Comment (AlexandrHoroshih) [reply src/solid/scope.ts:17]: Yep, that would be better move 🤔 

`getScope` should not be public, that's for sure

**Review by AlexandrHoroshih — CHANGES_REQUESTED** [review]
Thank you for your contribution! 🧡 

Everything looks good, all test in place, feature works as expected 🔥  👍 

There only few small caveats:

File: src/solid/scope.ts (hunk 2 above)
Comment (AlexandrHoroshih) [reply src/solid/scope.ts:62]: Hmm, i thougth a little more and now i suggest to do it like this:


1. Inside `getScope` there should be default fallback like `forceScope = true`

2. For `/scope` exports we don't need to do anything besides `getScope()` call without params

3. For main `effector-solid` exports we should do something like `getScope(ops?.forceScope ?? false)` inside

**Comment by usmanyunusov** [comment]
I don't pass the `scopes.test.tsx` solid test . What is the right way to test?

**Comment by AlexandrHoroshih** [comment]
Hmm, @Drevoed maybe you can help? 🤔 

I don't quite understand, what is happening in the tests and why changes in PR would affect this

**Comment by AlexandrHoroshih** [comment]
@usmanyunusov nevermind, i got it.

You moved hooks into `ssr.ts`, while they were placed at `scope.ts` - it has broken the build

I suggest to revert these changes (get hooks back into `scope.ts`) and move `getScope` and related Scope context into file named like `scope-context`

**Comment by AlexandrHoroshih** [comment]
`ssr.ts` would not work with Solid.js, because it never had such sub-export

There is one in React bindings, but it is a leftover of past times, when there were no other usecases for fork-api, than ssr

Now it is deprecated, basically, as `/scope` is much better naming now

In solid.js bindings it was done this way from the start

**Comment by usmanyunusov** [comment]
@AlexandrHoroshih, Fixed

**Review by AlexandrHoroshih — APPROVED** [review]
Looks good!

@Drevoed what you think?

**Comment by Drevoed** [comment]
LGTM, awesome work! Thanks :) :rocket:

**Pull request merged**